data-layers/data-layer-api-caller/src/main/java/com/flipkart/aesop/apicallerdatalayer/delete/ApiCallerDeleteDataLayer.java

```diff
@@ -0,0 +1,71 @@
+package com.flipkart.aesop.apicallerdatalayer.delete;
+
+import com.flipkart.aesop.destinationoperation.DeleteDestinationStoreProcessor;
+import com.flipkart.aesop.event.AbstractEvent;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.linkedin.databus.client.pub.ConsumerCallbackResult;
+import org.json.JSONObject;
+import org.trpr.platform.core.impl.logging.LogFactory;
+import org.trpr.platform.core.spi.logging.Logger;
+
+import java.io.DataOutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Iterator;
+import java.util.Map;
+
+/**
+ * Created by aman.gupta on 12/08/15.
+ */
+public class ApiCallerDeleteDataLayer extends DeleteDestinationStoreProcessor {
+
+    private static final Logger LOGGER = LogFactory.getLogger(ApiCallerDeleteDataLayer.class);
+    private URL url;
+    private JSONObject headers;
+    public ApiCallerDeleteDataLayer(URL url, JSONObject headers) {
+        this.url = url;
```

Comment:
Define an interface called 
public interface ContextualHeaderProvider {
Map<String,String> getContextualHeaders(Map<String,Object> eventMap);
}

Have another constructor in APICallerDeleteDataLayer and take an instance of this object as a parameter and store it in a global variable.
In the delete() method, invoke this interface's method to get the custom headers and add it to the headers list.